streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -274,27 +294,23 @@ public StateStore getStore(final String name) {
     public void flush() {
         RuntimeException firstException = null;
         // attempting to flush the stores
-        if (!registeredStores.isEmpty()) {
+        if (!stores.isEmpty()) {
```

Comment:
Note originally we need to do https://issues.apache.org/jira/browse/KAFKA-8204 only because we may reinitialize state stores, now since we removed that logic we do not need to maintain the stores as ordered<optional> anymore. cc @vvcephei